src-tauri/src/lib.rs

```diff
@@ -1,32 +1,123 @@
-use http::HttpClientImpl;
-use tauri::{Builder, Manager};
+use async_trait::async_trait;
+use services::{
+    note_service::{LocalNoteService, NoteService, RemoteNoteService},
+    user_service::{LocalUserService, RemoteUserService, UserService},
+};
+use tauri::{App, Builder, Manager};
 use tauri_plugin_store::StoreExt;
+use utils::http::HttpClientImpl;
 
 mod commands;
-mod http;
-mod note;
-mod service;
-mod user;
-
-use note::note_service::LocalNoteService;
-use note::note_service::RemoteNoteService;
-use note::NoteManager;
-use user::user_service::LocalUserService;
-use user::user_service::RemoteUserService;
-use user::UserManager;
+mod models;
+mod services;
+mod utils;
+
+#[async_trait]
+trait State: Send + Sync {
+    fn note_service(&self) -> &NoteService;
+    fn user_service(&self) -> &UserService;
+}
+
+struct LocalState {
+    note_service: LocalNoteService,
+    user_service: LocalUserService,
+}
+
+impl LocalState {
+    fn new(app: &mut App) -> Self {
+        let notes_store = app.handle().store_builder("notes.bin").build();
+        let note_service = LocalNoteService::new(notes_store);
+
+        let user_store = app.handle().store_builder("users.bin").build();
+        let user_service = LocalUserService::new(user_store);
+
+        Self {
+            note_service,
+            user_service,
+        }
+    }
+}
+
+#[async_trait]
+impl State for LocalState {
+    fn note_service(&self) -> &NoteService {
+        &self.note_service
+    }
+
+    fn user_service(&self) -> &UserService {
+        &self.user_service
+    }
+}
+
+struct RemoteState {
+    note_service: RemoteNoteService,
+    user_service: RemoteUserService,
+}
+
+impl RemoteState {
+    fn new() -> Self {
+        let remote_note_service =
+            RemoteNoteService::new("/notes".to_string(), HttpClientImpl::default());
+
+        let remote_user_service =
+            RemoteUserService::new("/users".to_string(), HttpClientImpl::default());
```

Comment:
/notes and /users shouldn't be given as params, but initialized from inside the constructor.

It's fine if this is done in another issue.

Reply:
Indeed, a whole config struct will hold all of this.